modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_backend_ipc.c

```diff
@@ -128,14 +128,13 @@ static void spinel_packet_send_thread_fn(void *arg1, void *arg2, void *arg3)
 	while (true) {
 		k_sem_take(&send_sem, K_FOREVER);
 		struct ringbuffer *buf = &ring_buffer[rd_idx];
-		uint32_t expected_ret = buf->len;
 
 		LOG_DBG("Sending %u bytes from send thread", buf->len);
 		int ret = ipc_service_send(&ept, buf->data, buf->len);
 
 		rd_idx = get_rb_idx_plus_1(rd_idx);
 
-		if (ret != expected_ret) {
+		if (ret < 0) {
```

Comment:
This looks like a workaround for a problem in the icbmsg backend. According to the `ipc_service_send()` function description, it must return the number of sent bytes. If icbmsg backend does not comply with the `ipc_service()` protocol, it shall be fixed. This file should not contain a workaround.

With this workaround, you could miss a case when the backend sends only a part of the message returning a positive number lower than the expected one.

Reply:
I reverted the condition back to its original form and made the icbmsg return sent bytes instead of blocks.

Additionally, I made a verification inside `send_control_message` where verifying sent bytes makes more sense in icbmsg because those 3 bytes there are what we actually send.